app.rb

```diff
@@ -0,0 +1,150 @@
+require './teacher'
+require './student'
+require './book'
+require './rental'
+
+ACTIONS = {
+  1 => :list_all_books,
+  2 => :list_all_people,
+  3 => :create_person,
+  4 => :create_book,
+  5 => :create_rental,
+  6 => :list_rentals_by_person_id,
+  7 => :break
+}.freeze
+
+class App
+  attr_reader :books, :people
+
+  def initialize()
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def options
+    puts ''
+    puts 'Please choose an option by enterin a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def select
+    loop do
+      options
+      option = gets.chomp.to_i
+      action = ACTIONS[option]
+
+      if action == :break
+        puts 'Thank you for using this app!'
+        break
+      elsif action
+        send(action)
+      else
+        puts 'Error: Invalid number, try again'
+      end
+    end
+  end
+
+  def list_all_books
+    books.each do |book, _index|
+      puts("Title: #{book.title}, Author: #{book.author}")
+    end
+  end
+
+  def list_all_people
+    people.each do |person|
+      puts("[#{person.is_a?(Student) ? 'Student' : 'Teacher'}]
+      Name: #{person.name},
+      ID: #{person.id},
+      Age: #{person.age}")
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    number = gets.chomp.to_i
+
+    case number
+    when 1
+      print('Age: ')
+      age = gets.chomp
+      print('Name: ')
+      name = gets.chomp
+      print('Has parent permission? [Y/N]: ')
+      parent_permission_console = gets.chomp.capitalize
+      if parent_permission_console == 'N'
+        @people.push(Student.new(age, name: name, parent_permission: false))
+      elsif parent_permission_console == 'Y'
+        @people.push(Student.new(age, name: name, parent_permission: true))
+      else
+        puts 'Please select between [Y/N]'
+      end
+      puts('Person created successfully')
+    when 2
```

Comment:
- [ ] Great job working on the functionality so far but when I try to create a new student, I face this error with the wrong number of arguments (given 2, expected 1), so kindly check on this error and try to solve the problem to let the user be able to create a student ⭐🔥💯🙏🏻

- When I try to create a student  ⬇
![image](https://github.com/Javivy/school-library-ruby/assets/58553711/97029896-ec30-4f90-9a5b-918ad096bcc3)